akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -291,15 +310,21 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
           val maxBytesToSend = currentMaxFrameSize min connectionWindowLeft
           val frame = outStream.nextFrame(maxBytesToSend)
           outlet.push(frame)
-          connectionWindowLeft -= frame.payload.size
-
-          if (outStream.canSend) // if we still can send, then do
-            become(WaitingForNetworkToSendData(sendableOutstreams))
-          else if (sendableOutstreams.size == 1)
-            // now empty
-            become(Idle)
-          else
-            become(WaitingForNetworkToSendData(sendableOutstreams.filterNot(_ == chosenId))) // TODO: use Set instead
+          frame match {
+            case DataFrame(_, _, payload) ⇒ connectionWindowLeft -= payload.size
+            case _                        ⇒
+          }
+
+          outStream.endStreamIfPossible() match {
```

Comment:
I think that's a viable solution. Would be nice if the duplication with the above case could be removed.